axelor-contract/src/main/java/com/axelor/apps/contract/web/ContractLineController.java

```diff
@@ -59,4 +75,157 @@ public void createAnalyticDistributionWithTemplate(
+
+          AnalyticAxis analyticAxis = new AnalyticAxis();
+
+          for (AnalyticAxisByCompany axis :
+              Beans.get(AccountConfigService.class)
+                  .getAccountConfig(contract.getCompany())
+                  .getAnalyticAxisByCompanyList()) {
+            if (axis.getSequence() + 1 == i) {
+              analyticAxis = axis.getAnalyticAxis();
+            }
+          }
+
+          for (AnalyticAccount analyticAccount :
+              Beans.get(AnalyticAccountRepository.class).findByAnalyticAxis(analyticAxis).fetch()) {
+            analyticAccountList.add(analyticAccount.getId());
+          }
+
+          if (ObjectUtils.isEmpty(analyticAccountList)) {
+            response.setAttr(
+                "axis".concat(Integer.toString(i)).concat("AnalyticAccount"),
+                "domain",
+                "self.id IN (0)");
+          } else {
+            if (contract.getCompany() != null) {
+              String idList =
+                  analyticAccountList.stream()
+                      .map(Object::toString)
+                      .collect(Collectors.joining(","));
+
+              response.setAttr(
+                  "axis" + i + "AnalyticAccount",
+                  "domain",
+                  "self.id IN ("
+                      + idList
+                      + ") AND self.statusSelect = "
+                      + AnalyticAccountRepository.STATUS_ACTIVE
+                      + " AND (self.company is null OR self.company.id = "
+                      + contract.getCompany().getId()
+                      + ")");
+            }
+          }
+        }
+      }
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void createAnalyticAccountLines(ActionRequest request, ActionResponse response) {
+    try {
+      if (request.getContext().getParent() != null
+          && (Contract.class)
+              .equals(request.getContext().getParent().getParent().getContextClass())) {
+
+        ContractLine contractLine = request.getContext().asType(ContractLine.class);
+        Contract contract = request.getContext().getParent().getParent().asType(Contract.class);
+        if (contract != null
+            && Beans.get(MoveLineComputeAnalyticService.class)
+                .checkManageAnalytic(contract.getCompany())) {
+          contractLine =
+              Beans.get(ContractLineService.class)
+                  .analyzeContractLine(contractLine, contract, contract.getCompany());
+          response.setValue("analyticMoveLineList", contractLine.getAnalyticMoveLineList());
+        }
+      }
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void manageAxis(ActionRequest request, ActionResponse response) {
+    try {
+      Contract contract = null;
+
+      if (request.getContext().getParent() != null
+          && (Contract.class)
+              .equals(request.getContext().getParent().getParent().getContextClass())) {
```

Comment:
same issue as above, the check could be factorized inside a protected method of this controller